### PR TITLE
Prelude documentation update

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: src/Streamly/Prelude.hs [hunk 1]
```diff
@@ -11,12 +11,230 @@
...
+-- example is @2 + 1 = 3@ seconds:
+--
+-- >>> stream :: SerialT IO Int = delay 2 `consM` delay 1 `consM` nil
+-- >>> Stream.toList stream -- IO [Int]
+-- 2 sec
+-- 1 sec
+-- [2,1]
+--
+-- 'AsyncT' executes the actions concurrently, so the total delay is @max 2 1 =
+-- 2@ seconds:
```
Comment (adithyaov) [inline src/Streamly/Prelude.hs:92]: Does `@` span over multiple lines?

**Comment by adithyaov** [comment]
I went through the draft. Looks good. Please check if everything is rendered properly (as expected) on haddocks.

File: src/Streamly/Internal/Data/Stream/Ahead.hs [hunk 2]
```diff
@@ -53,6 +63,15 @@ import Prelude hiding (map)
...
 
+-- $setup
+-- >>> import Control.Concurrent (threadDelay)
+-- >>> :{
+--  delay n = do
+--      threadDelay (n * 1000000)   -- sleep for n seconds
+--      putStrLn (show n ++ " sec") -- print "n sec"
+--      return n                    -- IO Int
+-- :}
+
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Ahead.hs:74]: Is this what you were talking about? doctests don't pick this up from the previous stuff (defined in the module header)?
I'm assuming `$setup` is identified by doctests?
Why did not put `import qualified Streamly.Prelude as Stream` here?

Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Stream/Ahead.hs:74]: The full thing is in module header, including this, to show in haddock docs for the user. setup section has only minimal required. Alternatively we can put everything in setup and use `$setup` in one export section named "Running Examples" or something. I did this in PR 10009.

Comment (adithyaov) [reply src/Streamly/Internal/Data/Stream/Ahead.hs:74]: Of course, PR 10009.

File: src/Streamly/Internal/Data/Stream/Async.hs [hunk 3]
```diff
@@ -817,28 +884,120 @@ consMWAsync m r = fromStream $ K.yieldM m `wAsync` (toStream r)
...
 --
 -- /Since: 0.2.0 ("Streamly")/
+
+-- Scheduling details:
+--
+-- This is how the execution of the above example proceeds:
+--
+-- 1. The scheduler queue is initialized with @[S.fromList [1,2,3],
+-- (S.fromList [4,5,6]) \<> (S.fromList [7,8,9])]@ assuming the head of the
+-- queue is represented by the  rightmost item.
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Async.hs:931]: Head of the queue is the left most item?

File: src/Streamly/Internal/Data/Stream/Zip.hs [hunk 4]
```diff
@@ -70,6 +73,15 @@ import Prelude hiding (map, repeat, zipWith, errorWithoutStackTrace)
...
 #include "Instances.hs"
 
+-- $setup
+-- >>> import Control.Concurrent (threadDelay)
+-- >>> :{
+--  delay n = do
+--      threadDelay (n * 1000000)   -- sleep for n seconds
+--      putStrLn (show n ++ " sec") -- print "n sec"
+--      return n                    -- IO Int
+-- :}
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Zip.hs:83]: You want to put this part in the module header as well?

File: src/Streamly/Internal/Data/Stream/Serial.hs [hunk 5]
```diff
@@ -388,14 +351,37 @@ infixr 6 `wSerial`
...
+-- Scheduling Notes:
+--
+-- Note that evaluation of @a \`wSerial` b \`wSerial` c@ does not interleave
+-- @a@, @b@ and @c@ with equal priority.  This expression is equivalent to @a
+-- \`wSerial` (b \`wSerial` c)@, therefore, it fairly interleaves @a@ with the
+-- result of @b \`wSerial` c@.  For example, @Stream.fromList [1,2] \`wSerial`
+-- Stream.fromList [3,4] \`wSerial` Stream.fromList [5,6]@ would result in
+-- [1,3,2,5,4,6].  In other words, the leftmost stream gets the same scheduling
+-- priority as the rest of the streams taken together. The same is true for
+-- each subexpression on the right.
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Serial.hs:383]: Ah, this is a little different from `wAsync`

**Review by adithyaov — APPROVED** [review]
I went through `src/Streamly/Prelude.hs` before so just skimmed through it now.
The doc changes look good. Easier to read and follow through.

File: src/Streamly/Internal/Data/Stream/Ahead.hs (hunk 2 above)
Comment (adithyaov) [reply src/Streamly/Internal/Data/Stream/Ahead.hs:74]: Not a big deal, but, if possible we should do it here as well. Works as a good standard. Reduces duplication and other errors surrounding it.

**Comment by harendra-kumar** [comment]
Will make the changes in another PR.

**Pull request merged**